x509/src/certificate.rs

```diff
@@ -87,6 +88,41 @@ pub struct TbsCertificate<'a> {
     pub extensions: Option<crate::ext::Extensions<'a>>,
 }
 
+impl<'a> TbsCertificate<'a> {
+    /// Decodes a single extension
+    ///
+    /// Returns an error if multiple of these extensions is present. Returns
+    /// `Ok(None)` if the extension is not present. Returns a decoding error
+    /// if decoding failed. Otherwise returns the extension.
+    pub fn get<'b: 'a, T: Decodable<'a> + AssociatedOid>(
+        &'b self,
+    ) -> Result<Option<(bool, T)>, Error> {
+        let mut iter = self.filter::<T>().peekable();
+        if let Some(item) = iter.next() {
+            return match iter.peek() {
+                Some(..) => Err(ErrorKind::Failed.into()),
+                None => Ok(Some(item?)),
+            };
+        }
+
+        Ok(None)
```

Comment:
I think this would be marginally clearer without an early `return`:

```suggestion
        if let Some(item) = iter.next() {
            match iter.peek() {
                Some(..) => Err(ErrorKind::Failed.into()),
                None => Ok(Some(item?)),
            }
        } else {
            Ok(None)
        }
```

Reply:
Fixed. Though I swapped out the branches for a match statement.